pkgs/development/python-modules/mlx-lm/default.nix

```diff
@@ -0,0 +1,70 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  wheel,
+  jinja2,
+  mlx,
+  numpy,
+  protobuf,
+  pyyaml,
+  transformers,
+  sentencepiece,
+  pytestCheckHook,
+  writableTmpDirAsHomeHook,
+}:
+
+buildPythonPackage rec {
+  pname = "mlx-lm";
+  version = "0.26.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "ml-explore";
+    repo = "mlx-lm";
+    tag = "v${version}";
+    hash = "sha256-J69XIqsjQ4sQqhx+EkjKcVXVlQ4A4PGJvICSiCfoSOA=";
+  };
+
+  build-system = [
+    setuptools
+    wheel
+  ];
+
+  dependencies = [
+    jinja2
+    mlx
+    numpy
+    protobuf
+    pyyaml
+    transformers
+  ];
+
+  nativeCheckInputs = [
+    writableTmpDirAsHomeHook
+    pytestCheckHook
+    sentencepiece
+  ];
+
+  pythonImportsCheck = [
+    "mlx_lm"
+  ];
+
+  disabledTestPaths = [
+    # ModuleNotFoundError: No module named 'datasets'
+    "tests/test_datsets.py::TestDatasets::test_hf"
+    # RuntimeError: [metal_kernel] No GPU back-end.
+    "tests/test_models.py::TestModels::test_bitnet"
```

Comment:
```suggestion
    # Requires network access to huggingface.co
    "tests/test_datsets.py"
    "tests/test_generate.py"
    "tests/test_server.py"
    "tests/test_tokenizers.py"
    "tests/test_utils.py::TestUtils::test_convert"
    "tests/test_utils.py::TestUtils::test_load"
    "tests/test_utils_load_model.py"

    # RuntimeError: [metal_kernel] No GPU back-end.
    "tests/test_models.py::TestModels::test_bitnet"
```

Reply:
Right, I've added these changes, will run nixpkgs-review in the background